internal/github_internal/github.go

```diff
@@ -57,3 +57,52 @@ func GetPullRequestFiles(owner string, repo string, prNumber int) ([]string, err
 	}
 	return fileNames, nil
 }
+
+func GetRepositoryFilesRecursive(owner, repo string) ([]string, error) {
+	client := NewGithubClient()
+	var files []string
+
+	err := fetchFilesRecursive(client, owner, repo, "", &files)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching repository files: %w", err)
+	}
+
+	return files, nil
+}
+
+func fetchFilesRecursive(client *github.Client, owner, repo, path string, files *[]string) error {
+	contents, dirContents, _, err := client.Repositories.GetContents(context.Background(), owner, repo, path, nil)
+	if err != nil {
+		return err
+	}
+
+	if contents != nil {
+		*files = append(*files, contents.GetPath())
+		return nil
+	}
+
+	for _, item := range dirContents {
```

Comment:
3. Inside the `internal/github_internal/github.go` file, within the `fetchFilesRecursive` function, we are not handling the error that might occur during recursive function call. It could potentially cause error hiding or handling issues. An error returned by `fetchFilesRecursive` function should be checked.